WooCommerce/Classes/ViewRelated/Filters/FilterListViewController.swift

```diff
@@ -0,0 +1,314 @@
+enum FilterListValueSelectorConfig {
+    // Standard list selector with fixed options
+    case staticOptions(options: [FilterType])
+    // Example: Categories
+    case custom
+}
+
+/// Contains data for rendering a filter type row.
+struct FilterListCellViewModel: Equatable {
+    /// The title of the filter type.
+    let title: String
+
+    /// The user-facing value of the filter type.
+    let value: String
+}
+
+/// A type that can be used to filter a list of models.
+protocol FilterType {
+    /// The user-facing description of the filter value.
+    var description: String { get }
+
+    /// Whether the filter is set to a non-empty value.
+    var isActive: Bool { get }
+}
+
+/// Allows the user to filter a list of models with generic filters.
+/// The UI consists of a list of filters at the top and a Filter CTA at the bottom that is always visible to apply the filters.
+/// Tapping on a filter in the list navigates to a list of options for the filter.
+///
+final class FilterListViewController: UIViewController {
+
+    @IBOutlet private weak var navigationControllerContainerView: UIView!
+    @IBOutlet private weak var filterActionContainerView: UIView!
+
+    private let viewModel: FilterListViewModel
+    private let listSelectorCommand: FilterListSelectorCommand
+
+    private lazy var listSelector: ListSelectorViewController
+        <FilterListSelectorCommand, FilterListSelectorCommand.Model, FilterListSelectorCommand.Cell> = {
+            return ListSelectorViewController(command: listSelectorCommand) { [weak self] _ in }
+    }()
+
+    private var clearAllBarButtonItem: UIBarButtonItem?
+
+    private var cancellableSelectedFilterType: ObservationToken?
+    private var cancellableSelectedFilterValue: ObservationToken?
+
+    init(viewModel: FilterListViewModel) {
+        self.viewModel = viewModel
+        self.listSelectorCommand = FilterListSelectorCommand(data: viewModel.filterTypeViewModels)
+        super.init(nibName: "FilterListViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        cancellableSelectedFilterType?.cancel()
+        cancellableSelectedFilterValue?.cancel()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigation()
+        configureMainView()
+        configureChildNavigationController()
+        configureFilterActionContainerView()
+        observeListSelectorCommandItemSelection()
+        updateUI(numberOfActiveFilters: viewModel.filterTypeViewModels.numberOfActiveFilters)
+    }
+
+    // MARK: - Navigation
+    //
+    @objc func filterActionButtonTapped() {
+        viewModel.onFilterActionTapped()
+    }
+
+    @objc func dismissButtonTapped() {
+        viewModel.onDismissActionTapped()
+    }
+
+    @objc func clearAllButtonTapped() {
+        viewModel.onClearAllActionTapped()
```

Comment:
I think we should rename this to just `clearAll()` so it's more of an _instruction_ for the `ViewModel` what to do. If we name it as `onClearAllActionTapped`, it would sound like `FilterProductListViewModel` can pretty much do whatever it wants. 😄 There are probably times that we do that but I think, in this case, we should just be clear and say the instruction. What do you think? 

```suggestion
        viewModel.clearAll()
```